File: src/main/java/br/com/leonardoferreira/jirareport/client/IssueClient.java
package br.com.leonardoferreira.jirareport.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestParam;

/**
 * Created by lferreira on 3/26/18
 */
@FeignClient(name = "issue-client", url = "${jira.url}")
public interface IssueClient {

    @GetMapping("/rest/api/2/search?expand=changelog&maxResults=100")
    String findAll(@RequestHeader("Authorization") String token, @RequestParam("jql") String jql);

}

File: src/main/java/br/com/jiratorio/client/IssueClient.kt
package br.com.jiratorio.client

import br.com.jiratorio.client.config.JiraClientConfiguration
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestParam

@FeignClient(
    name = "issue-client",
    url = "\${jira.url}",
    configuration = [
        JiraClientConfiguration::class
    ]
)
interface IssueClient {

    @GetMapping("/rest/api/2/search?expand=changelog&maxResults=100")
    fun findByJql(@RequestParam("jql") jql: String): String

}
